### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by clickCA**
Ready to be review

### CODE REVIEW

File: REST/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# REST-API
+
+## Overview
+
+This application is built using Go and MySQL and utilizes the Gin web framework. It allows you to perform CRUD operations on books.
+
+## Requirements
+
+- Go (Version 1.15 or higher)
```
Comment (Nick-solich) [inline REST/README.md]: I using go 1.18 but cannot run this need to use only go 1.21

File: REST/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+go run main.go
+```
+
+The application should now be running, and you should be connected to your MySQL database.
+
+### **Using Swagger to Test API**
+
+To test the API using Swagger:
+
+**Navigate to Swagger UI**: Open your web browser and go to **`http://localhost:8080/swagger/index.html`**.
```
Comment (Nick-solich) [inline REST/README.md:48]: we can change this to be a link
[http://localhost:8080/swagger/index.html](http://localhost:8080/swagger/index.html)

File: REST/internal/auth/service.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	db *gorm.DB
+}
+
+// New creates a new authentication service instance.
+func New(db *gorm.DB) *AuthService {
+	return &AuthService{db: db}
+}
+
+// Register a new user.
+func (s *AuthService) Register(email, password, userType string) (*models.Base, error) {
```
Comment (Nick-solich) [inline REST/internal/auth/service.go:21]: image

I didn't know much but when I registered, I saw an empty field in db (Is there any way to update it later could we initiate it in the beginning (student_id) if it is no use then remove it)

File: REST/internal/auth/service.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+		base = professor.Base
+	default:
+		return nil, errors.New("invalid user type")
+	}
+
+	return &base, nil
+}
+
+// Login checks the credentials and returns the user if they're correct.
+func (s *AuthService) Login(email, password, userType string) (*models.Base, error) {
```
Comment (Nick-solich) [inline REST/internal/auth/service.go:55]: For now I have no problem for this but later when we develop front cookie will be useful later (create next task or do it now)

**Review by Nick-solich — CHANGES_REQUESTED** [review]
🛑 Have some points to concern but overall LGTM!

**Comment by sirawitch** [comment]
I think it's ok

File: REST/internal/auth/service.go (hunk 3 above)
Comment (clickCA) [reply REST/internal/auth/service.go:21]: This `student_id` initialize when model got auto migrated from models folder
I think that  
`student_id` must require before create review then i wont remove or make any change for this now (need to talk later about when the student user need to add the academic infomation)

**Pull request merged**